src/Service.Tests/Authentication/Helpers/WebHostBuilderHelper.cs

```diff
@@ -75,6 +74,12 @@ public static class WebHostBuilderHelper
 
                             services.AddSingleton(runtimeConfigProvider);
 
+                            // https://github.com/dotnet/aspnetcore/issues/53332#issuecomment-2091861884
+                            // AddRouting() adds required services that .NET8 does not add by default
+                            // Without this, .NET8 fails to resolve the required services for the middleware
+                            // and tests fail.
+                            services.AddRouting();
```

Comment:
Does doing this on .NET 6 explicitly cause any side effects?

Reply:
No observable side effects. The usage of `services.AddRouting()` adds relevant services to the DI container. This is slightly different from `app.UseRouting()` which adds middleware to the request execution pipeline.